sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/CodeGeneratorWithInterpretedFallback.scala

```diff
@@ -37,19 +37,22 @@ object CodegenObjectFactoryMode extends Enumeration {
  */
 abstract class CodeGeneratorWithInterpretedFallback[IN, OUT] extends Logging {
 
-  def createObject(in: IN): OUT = {
+  def createObject(in: IN): OUT =
+    createObject(in, subexpressionEliminationEnabled = false)
```

Comment:
Can we eliminate it? I think we can use `SQLConf.get` directly when we need to access the conf. This should be done in a different PR though.

Reply:
ah, it might be yes. I'll check in follow-up.

Reply:
can we do that first? I think that will be a small change and can also make this PR simpler and easier to review.

Reply:
I took some more look, seems it's not an actual config, but just a method parameter. Maybe we should remove this config in 3.0.

that said, let's keep your PR as it is.

Reply:
ok, I'll do that first in another pr.